packages/contracts/contracts/optimistic-ethereum/OVM/chain/OVM_CanonicalTransactionChain.sol

```diff
@@ -453,25 +453,27 @@ contract OVM_CanonicalTransactionChain is iOVM_CanonicalTransactionChain, Lib_Ad
 
         // We will sequentially append leaves which are pointers to the queue.
         // The initial queue index is what is currently in storage.
-        uint40 nextQueueIndex = getNextQueueIndex();
+        uint40 firstSubsequentQueueIndex = getNextQueueIndex();
 
+        // Note that the curContext is uninitialized for the first set of checks
         BatchContext memory curContext;
         for (uint32 i = 0; i < numContexts; i++) {
             BatchContext memory nextContext = _getBatchContext(i);
 
             if (i == 0) {
-                // Execute a special check for the first batch.
+                // Execute a special check for the first context.
                 _validateFirstBatchContext(nextContext);
             }
 
-            // Execute this check on every single batch, including the first one.
-            _validateNextBatchContext(
-                curContext,
-                nextContext,
-                nextQueueIndex,
-                queueRef
-            );
-
+            if(i < numContexts) {
+                // Execute this check on every single context, except the final one.
+                _validateConsecutiveBatchContexts(
+                    curContext,
```

Comment:
Renaming this from `_validateNextBatchContext` to `_validateConsecutiveBatchContexts` is important to note, because it reflects a real change in how the function, particularly the subcall to `_validateEnqueuesBetweenContexts` works (see below).

Reply:
Agreed, this is a much better name which reflects the new functionality.  Originally the function was mostly checking a single context which is where this bug came from in the first place 😅